lib/plugins/aws/package/compile/layers/index.js

```diff
@@ -93,14 +101,77 @@ class AwsCompileLayers {
 
     newLayerOutput.Value = { Ref: layerLogicalId };
 
+    const layerHashOutputLogicalId = this.provider.naming.getLambdaLayerHashOutputLogicalId(
+      layerName
+    );
+    const newLayerHashOutput = this.cfOutputLayerHashTemplate();
+    newLayerHashOutput.Value = sha;
+
+    const layerS3KeyOutputLogicalId = this.provider.naming.getLambdaLayerS3KeyOutputLogicalId(
+      layerName
+    );
+    const newLayerS3KeyOutput = this.cfOutputLayerS3KeyTemplate();
+    newLayerS3KeyOutput.Value = newLayer.Properties.Content.S3Key;
+
     _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Outputs, {
       [layerOutputLogicalId]: newLayerOutput,
+      [layerHashOutputLogicalId]: newLayerHashOutput,
+      [layerS3KeyOutputLogicalId]: newLayerS3KeyOutput,
     });
   }
 
+  compareWithLastLayer(layerName) {
+    const stackName = this.provider.naming.getStackName();
+    const layerHashOutputLogicalId = this.provider.naming.getLambdaLayerHashOutputLogicalId(
+      layerName
+    );
+
+    return this.provider
+      .request('CloudFormation', 'describeStacks', { StackName: stackName })
+      .then(data => {
+        const lastHash = _.find(data.Stacks[0].Outputs, {
+          OutputKey: layerHashOutputLogicalId,
+        });
+        const compiledCloudFormationTemplate = this.serverless.service.provider
+          .compiledCloudFormationTemplate;
+        const newSha = compiledCloudFormationTemplate.Outputs[layerHashOutputLogicalId].Value;
+
+        if (lastHash != null && lastHash.OutputValue === newSha) {
+          const layerS3keyOutputLogicalId = this.provider.naming.getLambdaLayerS3KeyOutputLogicalId(
+            layerName
+          );
+          const lastS3Key = _.find(data.Stacks[0].Outputs, {
+            OutputKey: layerS3keyOutputLogicalId,
+          });
+          compiledCloudFormationTemplate.Outputs[layerS3keyOutputLogicalId].Value =
+            lastS3Key.OutputValue;
+
+          const layerLogicalId = this.provider.naming.getLambdaLayerLogicalId(layerName);
+
+          const layerResource =
+            compiledCloudFormationTemplate.Resources[layerLogicalId] ||
+            compiledCloudFormationTemplate.Resources[`${layerLogicalId}${lastHash.OutputValue}`];
+          layerResource.Properties.Content.S3Key = lastS3Key.OutputValue;
+
+          const layerObject = this.serverless.service.getLayer(layerName);
+          layerObject.artifactAlreadyUploaded = true;
+          this.serverless.cli.log(`Layer ${layerName} is already uploaded.`);
+        }
+      })
+      .catch(e => {
+        if (e.message.indexOf('does not exist') > -1) {
+          return BbPromise.resolve();
+        }
+        throw e;
+      });
+  }
+
   compileLayers() {
     const allLayers = this.serverless.service.getAllLayers();
-    return BbPromise.each(allLayers, layerName => this.compileLayer(layerName));
+    return BbPromise.each(allLayers, layerName => {
```

Comment:
Let's not rely on `BbPromise` functions that do not have native counterparts (we plan dropping `bluebird` dependency, as soon as we drop support for Node.js v6, with v2)

Reply:
Fixed in https://github.com/serverless/serverless/pull/7680/commits/1aa3f01f4cb2e2103a73c8fc3f08e1cc9a00654d